src/prune_builds

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/python3 -u
+
+'''
+    This script is called by both the `build` and `prune` commands.
+'''
+
+import os
+import sys
+import json
+import shutil
+import argparse
+import datetime
+import tempfile
+import subprocess
+
+# Let's just hardcode this here for now
+DEFAULT_KEEP_LAST_N = 3
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--workdir", required=True, help="Path to workdir")
+parser.add_argument("--keep-last-n", type=int, default=DEFAULT_KEEP_LAST_N,
+                    help="Number of builds to keep (0 for all)")
+args = parser.parse_args()
+
+skip_pruning = (args.keep_last_n == 0)
+
+# first, regen builds.json
+
+builds = []
+builds_dir = os.path.join(args.workdir, "builds")
+with os.scandir(builds_dir) as it:
+    for entry in it:
+
+        # ignore non-dirs
+        if not entry.is_dir(follow_symlinks=False):
+            # those are really the only three non-dir things we expect there
+            if entry.name not in ['builds.json', 'latest']:
+                print(f"Ignoring non-directory {entry.name}")
+            continue
+
+        # ignore dirs missing meta.json
+        meta_file = os.path.join(entry.path, 'meta.json')
+        if not os.path.isfile(meta_file):
+            print(f"Ignoring directory {entry.name}")
+            continue
+
+        # collect dirs and timestamps
+        with open(meta_file) as f:
+            j = json.load(f)
+        t = datetime.datetime.strptime(j['ostree-timestamp'],
+                                       '%Y-%m-%dT%H:%M:%SZ')
```

Comment:
It looks like this [could be simpler](https://stackoverflow.com/a/3908349) - probably a good idea to avoid hardcoding the exact format.